compiler/il/OMRNode.cpp

```diff
@@ -2316,17 +2316,115 @@ OMR::Node::dontEliminateStores(bool isForLocalDeadStore)
 bool
 OMR::Node::isNotCollected()
    {
-   TR::Compilation * comp = TR::comp();
-   if (self()->getOpCode().hasSymbolReference() && self()->getSymbolReference()->getSymbol()->isNotCollected())
-      return true;
-   else if (comp->generateArraylets() &&
-            (self()->getOpCode().isArrayRef() &&
-             (self()->getFirstChild()->isNotCollected() || !self()->getFirstChild()->isInternalPointer())))
-      return true;
-   return false;
+   return !self()->computeIsCollectedReference();
    }
 
 
+/**
+ * \brief
+ *    Check if the node is an internal pointer.
+ *
+ * \return
+ *    Return true if the node is an internal pointer, false otherwise.
+ *
+ * \note
+ *    This query currently only works on opcodes that can have a pinning array pointer,
+ *    i.e. aiadd, aiuadd, aladd, aluadd. Suppport for other opcodes can be added if necessary.
```

Comment:
The original query only supported those opcodes and it assumed whoever uses it knows that. To make the new query less risky, this query currently keeps that assumption.  

We have two `hasPinningArrayPointer`, one is on `OMR::Node`, the other is on `OMR::ILOpCode`. It should be better to use the `OMR::Node` here since it checks the property of a node. It currently uses the `OMR::ILOpCode::hasPinningArrayPointer` internally. We can change it to have its own implementation if the two queries were to diverge.